src/main/kotlin/mate/academy/HelloWorldAnonymousClasses.kt

```diff
@@ -8,6 +8,34 @@ class HelloWorldAnonymousClasses {
     }
 
     fun sayHello(names: List<String>): List<String> {
-        // TODO: implement
+        val greetings = object : HelloWorldGreeting {
+            override fun greet(): String {
+                return "Hello world"
+            }
+
+            override fun greetSomeone(someone: String): String {
+                return "Hello $someone"
+            }
+        }
+
+        val result = mutableListOf<String>()
+
+        if (names.isEmpty()) {
+            result.add(greetings.greet())
```

Comment:
Code duplication is a bad practice.
Please refactor logic in case if 'names' is not empty.
Note that default greetings would be called anyway.
Also this link may help you
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/is-not-empty.html